src/java/us/kbase/mobu/tester/test/ModuleTesterFailHardTest.java

```diff
@@ -0,0 +1,113 @@
+        // force function to always return null
+        when(CallbackServer.getCallbackUrl(anyInt(),any(String[].class))).thenReturn(null);
+        Assert.assertNull(CallbackServer.getCallbackUrl(0, new String[1]));
+    }
+
+    @AfterClass
+    public static void tearDownModule() throws Exception {
+        if (cleanupAfterTests)
+            for (String moduleName : createdModuleNames)
+                try {
+                    deleteDir(moduleName);
+                } catch (Exception ex) {
+                    System.err.println("Error cleaning up module [" +
+                            moduleName + "]: " + ex.getMessage());
+                }
+    }
+
+    @After
+    public void afterText() {
+        System.out.println();
+    }
+
+    private static void deleteDir(String moduleName) throws Exception {
+        File module = new File(moduleName);
+        if (module.exists() && module.isDirectory())
+            FileUtils.deleteDirectory(module);
+    }
+
+    private void init(String lang, String moduleName) throws Exception {
+        deleteDir(moduleName);
+        createdModuleNames.add(moduleName);
+        ModuleInitializer initer = new ModuleInitializer(moduleName, token.getUserName(),
+                lang, false);
+        initer.initialize(true);
+    }
+
+    private int runTestsInDocker(String moduleName) throws Exception {
+        File moduleDir = new File(moduleName);
+        return runTestsInDocker(moduleDir, token);
+    }
+
+    public static int runTestsInDocker(File moduleDir, AuthToken token) throws Exception {
+        return runTestsInDocker(moduleDir, token, false);
+    }
+
+    public static int runTestsInDocker(File moduleDir, AuthToken token,
+                                       boolean skipValidation) throws Exception {
+        DockerClientServerTester.correctDockerfile(moduleDir);
+        File testCfgFile = new File(moduleDir, "test_local/test.cfg");
+        String testCfgText = ""+
+                "test_token=" + token.getToken() + "\n" +
+                "kbase_endpoint=" + TestConfigHelper.getKBaseEndpoint() + "\n" +
+                "auth_service_url=" + TestConfigHelper.getAuthServiceUrl() + "\n" +
+                "auth_service_url_allow_insecure=" +
+                TestConfigHelper.getAuthServiceUrlInsecure() + "\n";
+        FileUtils.writeStringToFile(testCfgFile, testCfgText);
+        int exitCode = new ModuleTester(moduleDir).runTests(ModuleBuilder.DEFAULT_METHOD_STORE_URL,
+                skipValidation, false);
+        System.out.println("Exit code: " + exitCode);
+        return exitCode;
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testPythonModuleExample() throws Exception {
+        System.out.println("Test [testPythonModuleExample]");
+        String lang = "python";
+        String moduleName = SIMPLE_MODULE_NAME + "Python";
+        init(lang, moduleName);
+        int exitCode = runTestsInDocker(moduleName);
```

Comment:
Is there an equivalent test for the module runner change?

Reply:
Currently there is no test the successful execution case of ModuleRunner.java. This test will need to be built before the failure case can be tested. I am trying to figure out how I can build a test for the successful execution case first.